### PR TITLE
[airlift] One last tutorial pass

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Few more touchups to tutorials and snippets based on a last run-through. Everything seemed to work except the migrated load csv step, which has some relative filepath trickiness that this PR fixes.

This PR also explicitly asks the user to work in the blank `definitions.py` file, which should hopefully remove another potential source of error.

### CODE REVIEW

**Comment by benpankow** [comment]
* **#24079** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24079?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 ## Example: Migrating an Airflow DAG to Dagster
 
+This example demonstrates how to migrate an Airflow DAG to Dagster using the `dagster-airlift` package. It contains code examples of how to peer, observe, and migrate assets from an Airflow DAG to Dagster.
+The below guide will walk through each of these steps in detail.
+
+## Example Structure
+
+```plaintext
+tutorial_example
+├── shared: Contains shared Python & SQL code used Airflow and migrated Dagster code
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/examples/tutorial-example/README.md:10]: neat af

File: examples/experimental/dagster-airlift/scripts/update_readme_snippets.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ def update_readme_snippets(readme_filepath_raw: str, *snippet_files_raw: str):
         snippet_contents = snippet_file.read_text()
         if MAIN in snippet_contents:
             snippet_contents = snippet_contents[: snippet_contents.index(MAIN)]
+        snippet_contents = snippet_contents.replace("\\n", "\\\\n")
```
Comment (benpankow) [inline examples/experimental/dagster-airlift/scripts/update_readme_snippets.py:36]: sooner we can move this to docs the better 😭

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_cacheable_assets_def.py [hunk 3]
```diff
@@ -249,7 +249,11 @@ def build_airflow_asset_from_specs(
         op_tags=tags,
     )
     def _asset(context):
-        raise Exception("This should never be called.")
+        raise Exception(
+            "Attempted to materialize a non-executable Airflow asset."
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_cacheable_assets_def.py:253]: 👍🏻

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md [hunk 4]
```diff
@@ -462,15 +473,111 @@ defs = build_defs_from_airflow_instance(
...
+def export_duckdb_to_csv_defs(spec: AssetSpec, args: ExportDuckDbToCsvArgs) -> Definitions:
+    @multi_asset(name=f"export_{args.table_name}", specs=[spec])
+    def _multi_asset() -> None:
+        export_duckdb_to_csv(args)
+
+    return Definitions(assets=[_multi_asset])
+
+
+def rebuild_customers_list_defs() -> Definitions:
+    asset_defs = Definitions.merge(
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/examples/tutorial-example/README.md:537]: can you name this `defs`

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md [hunk 5]
```diff
@@ -462,15 +473,111 @@ defs = build_defs_from_airflow_instance(
...
+                csv_path=Path(os.environ["TUTORIAL_EXAMPLE_DIR"]) / "customers.csv",
+                duckdb_path=Path(os.environ["AIRFLOW_HOME"]) / "jaffle_shop.duckdb",
+                duckdb_database_name="jaffle_shop",
+            ),
+        ),
+    )
+
+    rebuild_customers_list_schedule = ScheduleDefinition(
+        name="rebuild_customers_list_schedule",
+        target=AssetSelection.assets(*asset_defs.assets),  # type: ignore
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/examples/tutorial-example/README.md:566]: `target="*"` ?

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md (hunk 4 above)
Comment (schrockn) [reply examples/experimental/dagster-airlift/examples/tutorial-example/README.md:537]: or `merged_defs` or something

File: examples/experimental/dagster-airlift/examples/tutorial-example/README.md (hunk 5 above)
Comment (schrockn) [reply examples/experimental/dagster-airlift/examples/tutorial-example/README.md:566]: If you want to be explicit, then go through `defs.get_asset_graph()`

e.g. ` defs.get_asset_graph().all_asset_keys`

**Review by schrockn — DISMISSED** [review]
Just would like to get to resolution on `standalone.py`

File: examples/experimental/dagster-airlift/examples/tutorial-example/tutorial_example/dagster_defs/stages/standalone.py [hunk 6]
```diff
@@ -59,20 +59,19 @@ def rebuild_customers_list_defs() -> Definitions:
...
-                duckdb_schema="raw_data",
                 duckdb_database_name="jaffle_shop",
             ),
         ),
     )
 
     rebuild_customers_list_schedule = ScheduleDefinition(
         name="rebuild_customers_list_schedule",
-        target=AssetSelection.assets(*asset_defs.assets),  # type: ignore
+        target=AssetSelection.assets(merged_defs.get_asset_graph().all_asset_keys),  # type: ignore
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/examples/tutorial-example/tutorial_example/dagster_defs/stages/standalone.py:69]: whats up with the type ignore?

**Review by schrockn — APPROVED** [review]

File: examples/experimental/dagster-airlift/examples/tutorial-example/tutorial_example/dagster_defs/stages/standalone.py (hunk 6 above)
Comment (benpankow) [reply examples/experimental/dagster-airlift/examples/tutorial-example/tutorial_example/dagster_defs/stages/standalone.py]: dropped it - this does have to be list-ified though if you check the newest push

**Pull request merged**